module/users/helper.js

```diff
@@ -502,196 +516,263 @@ module.exports = class UsersHelper {
           programId // program Id we have to that will be validated
         );
-        console.log(JSON.stringify(targetedPrograms))
 
         // check current program is targeted or not
-        if(targetedPrograms.data.length === 0){
-          let solutionIds = []
-          let importedSurveys = await surveyService.userSurveys(token,programId)
-          importedSurveys = importedSurveys.result
-          let importedObservations = await surveyService.userObservations(token,programId)
-          importedObservations = importedObservations.result
-          let importedProjects = await improvementProjectService.importedProjects(token,programId); 
-          
-          if(importedProjects.data.length> 0 && (type ==="" || type === constants.common.IMPROVEMENT_PROJECT)){
-            importedProjects.data.forEach((importedProject) => {
-              solutionIds.push(importedProject.solutionInformation._id)
-            })
-          }
-          
-          if(importedSurveys.length> 0 && (type ==="" || type === constants.common.SURVEY)){
-            importedSurveys.forEach((surveys)=>{
-              solutionIds.push(surveys.solutionId)
-            })
-          }
-          if(importedObservations.length> 0 && (type ==="" || type === constants.common.OBSERVATION)){
-            importedObservations.forEach((observation)=>{
-              solutionIds.push(observation.solutionId)
-            })
+        if (targetedPrograms.data.length === 0) {
+          const solutionIds = [];
+          const getAllResources = [];
+
+          /**
+           * @function importedProjects
+           * @function userSurveys
+           * @function userObservations
+           *
+           * @param token string: userToken
+           * @param programId string: programId
+           *
+           * @returns {Promise}
+           */
+          // Creates an array of promises based on users Input
+          switch (type) {
+            case constants.common.IMPROVEMENT_PROJECT:
+              getAllResources.push(
+                improvementProjectService.importedProjects(token, programId)
+              );
+              break;
+            case constants.common.SURVEY:
+              getAllResources.push(surveyService.userSurveys(token, programId));
+              break;
+            case constants.common.OBSERVATION:
+              getAllResources.push(
+                surveyService.userObservations(token, programId)
+              );
+              break;
+            default:
+              getAllResources.push(
+                improvementProjectService.importedProjects(token, programId)
+              );
+              getAllResources.push(surveyService.userSurveys(token, programId));
+              getAllResources.push(
+                surveyService.userObservations(token, programId)
+              );
           }
+          //here will wait till all promises are resolved
+          const allResources = await Promise.all(getAllResources);
```

Comment:
It will handle. while retruing and we are passing success key and if it fails then also it will return that key as false and that will be used in validating. @aks30